builtin/providers/aws/resource_aws_wafregional_ipset.go

```diff
@@ -0,0 +1,162 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafRegionalIPSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafRegionalIPSetCreate,
+		Read:   resourceAwsWafRegionalIPSetRead,
+		Update: resourceAwsWafRegionalIPSetUpdate,
+		Delete: resourceAwsWafRegionalIPSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip_set_descriptors": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsWafRegionalIPSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafregionalconn
+
+	wr := newWafRegionalRetryer(conn)
+	out, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+		params := &waf.CreateIPSetInput{
+			ChangeToken: token,
+			Name:        aws.String(d.Get("name").(string)),
+		}
+		return conn.CreateIPSet(params)
+	})
+	if err != nil {
+		return err
+	}
+	resp := out.(*waf.CreateIPSetOutput)
+	d.SetId(*resp.IPSet.IPSetId)
+	return resourceAwsWafRegionalIPSetUpdate(d, meta)
+}
+
+func resourceAwsWafRegionalIPSetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafregionalconn
+
+	params := &waf.GetIPSetInput{
+		IPSetId: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetIPSet(params)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "WAFNonexistentItemException" {
+			log.Printf("[WARN] WAF IPSet (%s) not found, error code (404)", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	var IPSetDescriptors []map[string]interface{}
+
+	for _, IPSetDescriptor := range resp.IPSet.IPSetDescriptors {
+		IPSet := map[string]interface{}{
+			"type":  *IPSetDescriptor.Type,
+			"value": *IPSetDescriptor.Value,
+		}
+		IPSetDescriptors = append(IPSetDescriptors, IPSet)
+	}
+
+	d.Set("ip_set_descriptors", IPSetDescriptors)
```

Comment:
Would you mind adding error check here? We tend to avoid error checking for `d.Set()` of all primitive fields merely to avoid cluttering the code, but we tend to do it for lists and sets as we were bitten by not doing it many times in the past.